### PR TITLE
Logging Added to Preferences

### PR DESCRIPTION
**Pull request by Loafie**
Changes made to add logging to the settings panel and change the directory where log files are stored.

### CODE REVIEW

**Review by LaCuneta — CHANGES_REQUESTED** [review]
Overall this looks good, a couple small issues I left comments on in the code.  The rebase to hexy didn't quite work as a merge commit got added, but that's fine as I can squash this when I merge it in, or if you want you can try to get your commits re-applied without the merge commit.

There are some other updates for logging I thought of while reviewing this, but I'll handle those separately.

File: netlogo-gui/project/autogen/i18n/GUI_Strings_en.txt [hunk 1]
```diff
@@ -456,6 +456,7 @@ edit.hubnet.view.height = Height (pixels)
 tools.preferences = NetLogo Preferences
 tools.preferences.restartRequired = (restart required)
 tools.preferences.uiLanguage = User Interface Language:
+tools.preferences.loggingEnabled = Logging Enabled:
```
[the remark this answers is not in the document]
Comment (Loafie) [reply netlogo-gui/project/autogen/i18n/GUI_Strings_en.txt]: Done

File: netlogo-gui/src/main/app/App.scala [hunk 2]
```diff
@@ -2,8 +2,9 @@
 
 package org.nlogo.app
 
-import javax.swing.{ JOptionPane, JMenu }
+import javax.swing.{ JOptionPane, JMenu, JFrame }
 import java.awt.event.ActionEvent
+import java.util.prefs.{ Preferences => JavaPreferences }
```
[the remark this answers is not in the document]
Comment (Loafie) [reply netlogo-gui/src/main/app/App.scala]: Done

File: netlogo-gui/src/main/app/App.scala [hunk 3]
```diff
@@ -383,8 +384,9 @@ class App extends
 
     frame.addLinkComponent(new CompilerManager(workspace, world, tabs.codeTab))
     frame.addLinkComponent(listenerManager)
-
-    if (loggingConfigPath != null) {
+    val prefs = JavaPreferences.userRoot.node("/org/nlogo/NetLogo")
+    if (loggingConfigPath != null || prefs.get("loggingEnabled", "false").toBoolean ){
```
[the remark this answers is not in the document]
Comment (Loafie) [reply netlogo-gui/src/main/app/App.scala]: Done

**Comment by LaCuneta** [comment]
@Loafie Is this ready for re-review?  Can you sign the CLA as well (it's showing unsigned)?

**Comment by Loafie** [comment]
Ready for re-review.

**Review by LaCuneta — APPROVED** [review]

**Pull request merged**